autogluon/task/text_prediction/text_prediction.py

```diff
@@ -0,0 +1,429 @@
+    ngpus_per_trial = min(ngpus_per_trial, get_gpu_count())
+    assert nthreads_per_trial > 0 and ngpus_per_trial >= 0,\
+        'Invalid number of threads and number of GPUs.'
+    return {'num_cpus': nthreads_per_trial, 'num_gpus': ngpus_per_trial}
+
+
+def infer_eval_stop_log_metrics(problem_type,
+                                label_shape,
+                                eval_metric=None,
+                                stopping_metric=None):
+    """Infer the evaluate, stopping and logging metrics
+
+    Parameters
+    ----------
+    problem_type
+        Type of the problem
+    label_shape
+        Shape of the label
+    eval_metric
+        The eval metric provided by the user
+    stopping_metric
+        The stopping metric provided by the user
+
+    Returns
+    -------
+    eval_metric
+        The updated evaluation metric
+    stopping_metric
+        The updated stopping metric
+    log_metrics
+        The updated logging metric
+    """
+    if eval_metric is not None and stopping_metric is None:
+        stopping_metric = eval_metric
+        if isinstance(eval_metric, list):
+            stopping_metric = eval_metric[0]
+    if problem_type == _C.CLASSIFICATION:
+        if stopping_metric is None:
+            stopping_metric = 'acc'
+        if eval_metric is None:
+            eval_metric = 'acc'
+        if label_shape == 2:
+            log_metrics = ['f1', 'mcc', 'auc', 'acc', 'nll']
+        else:
+            log_metrics = ['acc', 'nll']
+    elif problem_type == _C.REGRESSION:
+        if stopping_metric is None:
+            stopping_metric = 'mse'
+        if eval_metric is None:
+            eval_metric = 'mse'
+        log_metrics = ['mse', 'rmse', 'mae']
+    else:
+        raise NotImplementedError
+    for other_log_metric in [stopping_metric, eval_metric]:
+        if isinstance(other_log_metric, str) and other_log_metric not in log_metrics:
+            log_metrics.append(other_log_metric)
+        else:
+            if isinstance(other_log_metric, list):
+                for ele in other_log_metric:
+                    if ele not in log_metrics:
+                        log_metrics.append(ele)
+    return eval_metric, stopping_metric, log_metrics
+
+
+@use_np
+class TextPrediction(BaseTask):
+    """AutoGluon Task for predicting labels based on text data."""
+    Dataset = tabular_prediction.TabularDataset
+
+    @classmethod
+    def fit(cls, train_data,
+            label,
+            tuning_data=None,
+            time_limits=None,
+            output_directory='./ag_text',
+            feature_columns=None,
+            holdout_frac=0.15,
+            eval_metric=None,
+            stopping_metric=None,
+            nthreads_per_trial=None,
+            ngpus_per_trial=None,
+            dist_ip_addrs=None,
+            scheduler=None,
+            num_trials=None,
+            reduction_factor=None,
+            grace_period=None,
+            max_t=None,
```

Comment:
Currently, the design is that the user can either specify the arguments by explicitly setting the kwargs, e.g., `max_t=40`,  or specify them via hyperparameters.

Reply:
But we may change that later.

Reply:
This could get confusing if they specify 2 different values of `max_t` in both kwargs and `hyperparameters`. I think just having 1 way to specify HPO arguments should be good enough.